src/hardware/siglent-sds/api.c

```diff
@@ -429,6 +437,36 @@ static int config_get(uint32_t key, GVariant **data,
 			*data = g_variant_new_string("History");
 		break;
 	case SR_CONF_SAMPLERATE:
+		/* SDS2000X+: The channels need to be enabled/disabled prior to
+		 * obtaining the sample rate when the data source is set to display.
+		 * If a channel is turned on/off on the oscilloscope then use Sigrok's settings instead. */
+		if (!strcmp(devc->model->series->name, "SDS2000X+") && devc->data_source == DATA_SOURCE_SCREEN) {
```

Comment:
I prefer avoiding to test things like model name strings for feature support... adding a bool flag (or bit-masked enums, I would look at other more mature drivers to see if there is a more common pattern) would be preferable esp. if future models also need this